viewer/src/logic/buttons/setup.ts

```diff
@@ -0,0 +1,88 @@
+import { AvailableSetupOption, Booster, Command, Event, SetupType, TechTile, tiles } from "@gaia-project/engine";
+import { ButtonData } from "../../data";
+import { eventDesc } from "../../data/event";
+import { federationData } from "../../data/federations";
+import { roundScoringNames } from "../../data/round-scorings";
+import { finalScoringSources } from "../charts/final-scoring";
+import { symbolButton, textButton } from "../commands";
+
+function chooseTechButton(command: string, data: AvailableSetupOption, label: string) {
+  const button = textButton({
+    command,
+    label,
+    buttons: data.options.map((tile) =>
+      symbolButton({
+        command: tile,
+        tech: { tile: tile as TechTile, commandOverride: tile },
+      })
+    ),
+  });
+  button.buttons[0].onCreate = (controller) => controller.subscribeButtonClick("techClick");
```

Comment:
Rather than assigning `onCreate` (and `onClick`) with callbacks, I'd rather make them event emitters / listen to events, but I should probably have made that comment a few PRs ago 😅

Reply:
not entirely sure what you mean, let's take this discussion to a different PR